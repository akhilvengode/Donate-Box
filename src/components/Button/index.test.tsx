import MyButton from ".";
import { shallow } from "enzyme";

describe("Testing Button component", () => {
  const onClickMock = jest.fn();
  const wrapper = shallow(
    <MyButton id="test" type="button" onClick={onClickMock}>
      Hello
    </MyButton>
  );

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render the button correctly", () => {
    expect(wrapper).toHaveLength(1);
  });

  it("should call the function when clicked", () => {
    wrapper.simulate("click");
    expect(onClickMock).toBeCalled();
  });
});
