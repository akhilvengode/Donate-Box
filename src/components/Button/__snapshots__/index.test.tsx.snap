// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Button component should match the snapshot 1`] = `
<button
  className="button"
  id="test"
  onClick={[MockFunction]}
  type="button"
>
  Hello
</button>
`;
